woocommerce/facebook-for-woocommerce
# Remove sync not possible modal.

### Changes proposed in this Pull Request:



Closes #2636

In this PR I remove the modal that warns the merchant before crating a product if certain conditions are met. The modal was shown during the creation or update action of a product when:
- that product was set to sync
- product had a category assigned and that category was excluded from Facebook sync.

The original modal was intended to warn the merchant about the excluding sync conditions. The interaction looked as follows:

https://github.com/woocommerce/facebook-for-woocommerce/assets/17271089/a4152e39-33c1-4b33-a72d-309457df4cfe

The intended behaviour was to "force" merchant to change the settings and make the product setup consistent. But this has backfired and is causing merchants a lot of problems. The product that is using the excluded from sync category can't be updated in any way.

The simplest fix is to remove the modal entirely. This solution is supported by the fact that the Facebook Product sync section will show what is blocking the sync for that product so no information is being lost. This feature has been implemented some time ago and looks like this:

<img src="https://github.com/woocommerce/facebook-for-woocommerce/assets/17271089/035f4708-f48f-4bd6-8995-c63fdf43e124" width="300">


### Screenshots:




### Detailed test instructions:


1. Create a product category
2. Go to `Marketing > Facebook > Product sync > Exclude categories from sync` and add that category to the excluded list
3. Create a new product and assign the excluded category to that product
4. Try to publish/update that product - it should work
5. Compare to the `develop` (rebuild required) branch where this operation should fail


### Additional details:

The modal was displayed in two cases:
1. When the product was created or updated
2. When the merchant was opening a product editor for a product with the excluded setup

I am removing modal for both cases. For the case 2 the modal was never displayed due to a bug.

The `<script>` added to the footer was doing an AJAX call with a wrong parameter

https://github.com/woocommerce/facebook-for-woocommerce/blob/c4c7b6a52d4ed66f6a0f497a9bd6ba2d79a5f7fd/includes/Admin.php#L1553

when a different name was expected by the AJAX handler:

https://github.com/woocommerce/facebook-for-woocommerce/blob/c4c7b6a52d4ed66f6a0f497a9bd6ba2d79a5f7fd/includes/AJAX.php#L177

This makes the total removal of the popup modal safe.


### Changelog entry

> Fix - Remove popup modal blocking product edit when product is using category excluded from sync.
